quic/s2n-quic-tls/src/callback.rs

```diff
@@ -349,10 +352,12 @@ pub struct State {
 impl State {
     /// Complete the handshake
     pub fn on_handshake_complete(&mut self) {
-        debug_assert_eq!(self.tx_phase, HandshakePhase::Handshake);
-        debug_assert_eq!(self.rx_phase, HandshakePhase::Handshake);
-        self.tx_phase.transition();
-        self.rx_phase.transition();
+        if self.rx_phase == HandshakePhase::Handshake {
+            self.rx_phase.transition();
+        }
```

Comment:
Should we keep the rx side the same?

Reply:
I'm not sure I know what you mean by this. Like, not transition the rx side from Handshake to Application?

Reply:
Keep it the same as it was previously where we do the debug assert and transition regardless